presto-main/src/main/java/com/facebook/presto/metadata/FunctionRegistry.java

```diff
@@ -969,7 +969,7 @@ public static OperatorType unmangleOperator(String mangledName)
     public boolean isMoreSpecificThan(ApplicableFunction left, ApplicableFunction right)
     {
         List<Type> resolvedTypes = resolveTypes(left.getBoundSignature().getArgumentTypes(), typeManager);
-        Optional<BoundVariables> boundVariables = new SignatureBinder(typeManager, right.getDeclaredSignature(), true)
+        Optional<BoundVariables> boundVariables = new DefaultSignatureBinder(typeManager, right.getDeclaredSignature(), true)
```

Comment:
I think you should use `signatureBinder` from the right `ApplicableFunction`. Maybe `ApplicableFunction` should contain `SqlFunction` or `signatureBinder` field in addition to `declaredSignature`.